src/librbd/cache/pwl/rwl/WriteLog.cc

```diff
@@ -319,6 +318,7 @@ bool WriteLog<I>::initialize_pool(Context *on_finish, pwl::DeferredContexts &lat
       this->m_free_log_entries = 0;
       lderr(cct) << "failed to initialize pool (" << this->m_log_pool_name << ")" << dendl;
       on_finish->complete(-pmemobj_tx_errno());
+      pmemobj_close(m_log_pool);
       return false;
```

Comment:
nit: I would introduce `int r = -EINVAL` variable and store `-pmemobj_tx_errno()` there, use `on_finish->complete(r)` in `error_handle`, so  could use `goto error_handle` here too to have it more uniform.